ElectronClient/app/gui/Header.jsx

```diff
@@ -75,6 +75,17 @@ class HeaderComponent extends React.Component {
 		if (nextProps.windowCommand) {
 			this.doCommand(nextProps.windowCommand);
 		}
+
+		if (nextProps.zoomFactor !== this.props.zoomFactor || nextProps.size !== this.props.size) {
+			const mediaQuery = window.matchMedia(`(max-width: ${550 * nextProps.zoomFactor}px)`);
+			const showButtonLabels = !mediaQuery.matches;
+
+			if (this.state.showButtonLabels !== showButtonLabels) {
+				this.setState({
+					showButtonLabels: !mediaQuery.matches,
+				});
+			}
+		}
```

Comment:
For new code, I think we shouldn't use `componentWillReceiveProps` as it is deprecated. I think in this case you can simply use `componentDidUpdate` and compare `prevProps` to `this.props`